### ISSUE TITLE
Clone error on CSSUTILS

### ISSUE THREAD

**Issue by viskey98** [mushorg/snare#113]
I tried cloning my Institute's Intranet website, and faced many errors in CSS utils.
I captured the errors in an error log and this is what i found:
```shell

2018-03-12 12:52:49 ERROR:CSSUTILS:__handle: Property: Invalid value for "CSS Level 2.1" property: -webkit-gradient(linear, left top, left bottom, color-stop(0%, #0272a7), color-stop(100%, #013953)) [44:2: background]
2018-03-12 12:52:49 ERROR:CSSUTILS:__handle: Property: Invalid value for "CSS Level 2.1" property: -webkit-linear-gradient(top, #FFF 0%, #FFF 100%) [45:2: background]
2018-03-12 12:52:49 ERROR:CSSUTILS:__handle: Property: Invalid value for "CSS Level 2.1" property: -o-linear-gradient(top, #FFF 0%, #FFF 100%) [46:2: background]
2018-03-12 12:52:49 ERROR:CSSUTILS:__handle: Property: Invalid value for "CSS Level 2.1" property: -ms-linear-gradient(top, #FFF 0%, #FFF 100%) [47:2: background]
2018-03-12 12:52:49 ERROR:CSSUTILS:__handle: Property: Invalid value for "CSS Level 2.1" property: linear-gradient(top, #FFF 0%, #FFF 100%) [48:2: background]
2018-03-12 12:52:49 ERROR:CSSUTILS:__handle: Property: Invalid value for "CSS Level 2.1" property: 16x [350:2: font-size]
2018-03-12 12:52:49 ERROR:CSSUTILS:__handle: Property: Invalid value for "CSS Level 2.1" property: -webkit-gradient(linear, 0% 0%, 0% 100%, from(#EEE), to(#BBB)) [423:2: background]
2018-03-12 12:52:50 ERROR:CSSUTILS:__handle: Property: Invalid value for "CSS Level 2.1" property: -webkit-gradient(linear, left top, left bottom, from(#1490ac), to(#006c82)) [113:2: background]
2018-03-12 12:52:50 ERROR:CSSUTILS:__handle: Property: Invalid value for "CSS Level 2.1" property: -webkit-linear-gradient(top, #1490ac, #006c82) [114:2: background]
2018-03-12 12:52:50 ERROR:CSSUTILS:__handle: Property: Invalid value for "CSS Level 2.1" property: -moz-linear-gradient(top, #1490ac, #006c82) [115:2: background]
2018-03-12 12:52:50 ERROR:CSSUTILS:__handle: Property: Invalid value for "CSS Level 2.1" property: -ms-linear-gradient(top, #1490ac 0, #006c82 100%) [118:2: background-image]
2018-03-12 12:52:50 ERROR:CSSUTILS:__handle: Property: Invalid value for "CSS Backgrounds and Borders Module Level 3" property: rgba(0, 0, 0, 0.3) 0 1px 1px [121:2: box-shadow]
2018-03-12 12:52:50 ERROR:CSSUTILS:__handle: Property: Invalid value for "CSS Text Level 3" property: #333 0 1px 0 [123:95: text-shadow]
2018-03-12 12:52:50 ERROR:CSSUTILS:__handle: Property: Invalid value for "CSS Level 2.1" property: -webkit-gradient(linear, left top, left bottom, from(#1490ac), to(#006c82)) [173:2: background]
2018-03-12 12:52:50 ERROR:CSSUTILS:__handle: Property: Invalid value for "CSS Level 2.1" property: -webkit-linear-gradient(top, #1490ac, #006c82) [174:2: background]
2018-03-12 12:52:50 ERROR:CSSUTILS:__handle: Property: Invalid value for "CSS Level 2.1" property: -moz-linear-gradient(top, #1490ac, #006c82) [175:2: background]
2018-03-12 12:52:50 ERROR:CSSUTILS:__handle: Property: Invalid value for "CSS Level 2.1" property: -ms-linear-gradient(top, #1490ac 0, #006c82 100%) [178:2: background-image]
2018-03-12 12:52:50 ERROR:CSSUTILS:__handle: Property: Invalid value for "CSS Backgrounds and Borders Module Level 3" property: rgba(0, 0, 0, 0.3) 0 1px 1px [181:2: box-shadow]
2018-03-12 12:52:50 ERROR:CSSUTILS:__handle: Property: Invalid value for "CSS Text Level 3" property: #333 0 1px 0 [188:2: text-shadow]
2018-03-12 12:52:50 ERROR:CSSUTILS:__handle: Property: Invalid value for "CSS Text Level 3" property: #333 0 1px 0 [291:91: text-shadow]

```
**I think cloner is not able to handle some CSS utils.**

**Comment by mzfr** [comment]
@viskey98 Can you provide the link for the website

**Comment by viskey98** [comment]
@afeena what do you think about this issue?

**Comment by afeena** [comment]
@viskey98 Could you please give the link to this website?

**Comment by viskey98** [comment]
It's intranet website, it is accessible only from inside the network here

**Comment by mzfr** [comment]
@afeena  @viskey98 
The error seems to be checking for some type of validation for the CSS so  we can try to change        [the following](https://github.com/mushorg/snare/blob/master/clone.py#L165) to    `css =   cssutils.parseString(data, validate=false)`.

I am not sure about this since I couldn't try this myself.

Also in my opinion these errors should be ignored during the parse call because according to [docs](https://pythonhosted.org/cssutils/docs/parse.html#parsestring) `validate=None` by default

**Comment by viskey98** [comment]
@mzfr Yah you are right, I also figured out these are just minor errors and need to be ignored. I found no major issue in the cloned page. Thanks for your help :)

**Comment by mzfr** [comment]
@viskey98 have you tried it with using `css = cssutils.parseString(data, validate=false)` ?

**Comment by viskey98** [comment]
Yes, it had removed those errors from the log. As, it doesn't interfere in the cloning process, I guess we can just ignore them. :)

**Comment by mzfr** [comment]
so I think we should do it in our main code base too so it doesn't bother anyone else ?

**Comment by viskey98** [comment]
We can. But, I think its better to wait until someone else gets the same error in some website. As I haven't faced this validation error in any website except this intranet site. And moreover SNARE is mean't for web applications. I will keep this open for discussion.

**Comment by mzfr** [comment]
@viskey98 One another suggestion I can give is that you can check what is a different thing does your website have comparing to other websites for example [github](https://github.com/) or any other website.
(I am talking in context of CSS) Maybe you get something good

**Comment by glaslos** [comment]
How about adding a command line flag to the cloner to disable the verification?

**Comment by viskey98** [comment]
@glaslos That sounds great, I will implement it :)

**Comment by viskey98** [comment]
Fixed by #120
